cmd/lifecycle/analyzer.go

```diff
@@ -20,62 +21,88 @@ import (
 
 type analyzeCmd struct {
 	//flags: inputs
-	cacheDir      string
-	cacheImageTag string
-	groupPath     string
-	uid, gid      int
 	analyzeArgs
+	uid, gid int
 
 	//flags: paths to write data
 	analyzedPath string
 }
 
 type analyzeArgs struct {
-	//inputs needed when run by creator
-	imageName  string
-	layersDir  string
-	skipLayers bool
-	useDaemon  bool
-
-	platform cmd.Platform
+	imageName   string
+	layersDir   string
+	orderPath   string //nolint - Platform API >= 0.7
+	runImageRef string //nolint - Platform API >= 0.7
+	stackPath   string //nolint - Platform API >= 0.7
+	useDaemon   bool
+
+	platform06     analyzeArgsPlatform06
+	additionalTags cmd.StringSlice        //nolint Platform API >= 0.7
+	docker         client.CommonAPIClient // construct if necessary before dropping privileges
+	keychain       authn.Keychain
+	platform       cmd.Platform
+}
 
-	//construct if necessary before dropping privileges
-	docker   client.CommonAPIClient
-	keychain authn.Keychain
+type analyzeArgsPlatform06 struct {
+	cacheDir      string
+	cacheImageTag string
+	groupPath     string
+	skipLayers    bool
+	cache         lifecycle.Cache
+	group         buildpack.Group
 }
 
 func (a *analyzeCmd) DefineFlags() {
 	cmd.FlagAnalyzedPath(&a.analyzedPath)
-	cmd.FlagCacheDir(&a.cacheDir)
-	cmd.FlagCacheImage(&a.cacheImageTag)
-	cmd.FlagGroupPath(&a.groupPath)
 	cmd.FlagLayersDir(&a.layersDir)
-	cmd.FlagSkipLayers(&a.skipLayers)
+	if a.platformAPIVersionGreaterThan06() {
+		cmd.FlagOrderPath(&a.orderPath)
+		cmd.FlagPreviousImage(&a.imageName)
+		cmd.FlagRunImage(&a.runImageRef)
+		cmd.FlagStackPath(&a.stackPath)
+		cmd.FlagTags(&a.additionalTags)
+	} else {
+		cmd.FlagCacheImage(&a.platform06.cacheImageTag)
```

Comment:
I think we'll want to move `cacheImageTag` out of `platform06` when we do https://github.com/buildpacks/lifecycle/issues/572 - but probably it could live here for now.